File: ecommerce-api-legacy/src/middlewares/errorHandler.js
const logger = require('../utils/logger');

class AppError extends Error {
    constructor(message, status = 400) {
        super(message);
        this.status = status;
        this.expose = true;
    }
}

function errorHandler(err, req, res, _next) {
    if (err instanceof AppError) {
        return res.status(err.status).json({ error: err.message });
    }
    logger.error('unhandled.exception', { path: req.path, message: err.message });
    res.status(500).json({ error: 'Internal Server Error' });
}

module.exports = { AppError, errorHandler };

File: task-manager-api/middlewares/error_handler.py
from flask import jsonify


class AppError(Exception):
    def __init__(self, message, status=400):
        super().__init__(message)
        self.message = message
        self.status = status


def register_error_handlers(app):
    @app.errorhandler(AppError)
    def handle_app_error(err):
        return jsonify({'error': err.message}), err.status

    @app.errorhandler(404)
    def handle_404(_err):
        return jsonify({'error': 'Recurso não encontrado'}), 404

    @app.errorhandler(405)
    def handle_405(_err):
        return jsonify({'error': 'Método não permitido'}), 405

    @app.errorhandler(Exception)
    def handle_generic(err):
        app.logger.exception('Unhandled exception: %s', err)
        return jsonify({'error': 'Erro interno do servidor'}), 500
